agents/PersonBirthAgent.java

```diff
@@ -0,0 +1,73 @@
+package grakn.simulation.agents;
+
+import grakn.client.GraknClient.Transaction;
+import grakn.client.answer.ConceptMap;
+import grakn.simulation.common.RandomSource;
+import graql.lang.Graql;
+import graql.lang.query.GraqlInsert;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Random;
+
+import static java.util.stream.Collectors.toList;
+
+public class PersonBirthAgent implements CityAgent {
+
+    @Override
+    public void iterate(AgentContext context, RandomSource randomSource, World.City city) {
+
+        List<String> femaleForenames = context.getWorld().getFemaleForenames().map(World.FemaleForename::getValue).collect(toList());
+        List<String> maleForenames = context.getWorld().getMaleForenames().map(World.MaleForename::getValue).collect(toList());
+        List<String> surnames = context.getWorld().getSurnames().map(World.Surname::getValue).collect(toList());
```

Comment:
I'd look to provide a way to get or generate a single name from the list via world rather than collecting all names locally each iteration. e.g. a `getWorld().getRandomForename(Random random);` Or return an `UnmodifiableList` of the names to do this locally.

Reply:
Good catch! I've gone with the `UnmodifiableList` option, anticipating that providing should always be the easy route

Reply:
To remove needless iterations over the names, right @adammitchelldev ?

I agree with the idea. I would do it in this way so we don't even need to create a helper function:

```
List<String> maleForenames = context.getWorld().getMaleForenames().map(World.MaleForename::getValue).collect(toList());

...

forename = maleForenames.get(random.nextInt(maleForenames.size()));
```

into this
```
List<...> maleForenames = context.getWorld().getMaleForenames();

...

forename = maleForenames.get(random.nextInt(maleForenames.size())).getValue();
```

Reply:
Have a look at the way this is addressed in the commit I just pushed and see if you're happy